internals/controllers/updatequota_controller.go

```diff
@@ -237,24 +234,33 @@ func subSnsQuota(sns *utils.ObjectContext, quotaSpec corev1.ResourceQuotaSpec) e
 		return err
 	}
 
-	// ensure quota is equal to sns
-	err = ensureSnsEqualQuota(sns.Object.(*danav1.Subnamespace).Spec.ResourceQuotaSpec, utils.GetSnsQuotaSpec(sns.Object))
+	err = ensureSnsEqualQuota(sns)
 	if err != nil {
 		return err
 	}
-	time.Sleep(400 * time.Millisecond)
-
 	return nil
 }
 
-func ensureSnsEqualQuota(snsQuota corev1.ResourceQuotaSpec, quota corev1.ResourceQuotaSpec) error {
-	for ok := false; ok; ok = false {
-		ok = true
-		for res := range snsQuota.Hard {
-			if snsQuota.Hard[res] != quota.Hard[res] {
+// ensureSnsEqualQuota compares the sns quota spec and the resource quota spec
+// in a loop until they are equal. this way we can know that the subnamespace
+// has been properly updated before doing the updatequota operation
+func ensureSnsEqualQuota(sns *utils.ObjectContext) error {
+	ok := false
+	quotaObj, err := utils.GetSNSQuotaObj(sns)
+	if err != nil {
+		return err
+	}
+	resourceQuotaSpec := utils.GetQuotaObjSpec(quotaObj.Object)
```

Comment:
This does need to be inside the while loop since we need to fetch the new object everytime to make sure it's been updated properly